controller/controlplane/manager_options_test.go

```diff
@@ -754,3 +755,50 @@ func TestManagerConfigToStatusFeatureGates(t *testing.T) {
 		})
 	}
 }
+
+func TestWithAnonymousReports(t *testing.T) {
+	testCases := []struct {
+		name     string
+		enabled  bool
+		expected bool
+	}{
+		{
+			name:     "anonymous reports enabled",
+			enabled:  true,
+			expected: true,
+		},
+		{
+			name:     "anonymous reports disabled",
+			enabled:  false,
+			expected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			cfg := &managercfg.Config{}
+			opt := WithAnonymousReports(tc.enabled)
+			opt(cfg)
+			assert.Equal(t, tc.expected, cfg.AnonymousReports)
+		})
+	}
+}
+
+func TestWithAnonymousReportsFixedPayloadCustomizer(t *testing.T) {
+	called := false
+	mockCustomizer := func(payload telemetry.Payload) telemetry.Payload {
+		called = true
+		payload["test"] = "value"
+		return payload
+	}
+
+	cfg := &managercfg.Config{}
+	opt := WithAnonymousReportsFixedPayloadCustomizer(mockCustomizer)
+	opt(cfg)
+
+	assert.NotNil(t, cfg.AnonymousReportsFixedPayloadCustomizer)
```

Comment:
```suggestion
	require.NotNil(t, cfg.AnonymousReportsFixedPayloadCustomizer)
```

if `cfg.AnonymousReportsFixedPayloadCustomizer` is `nil` line 801 below will panic